src/App.tsx

```diff
@@ -1,204 +1,92 @@
-/* eslint-disable jsx-a11y/label-has-associated-control */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import { USER_ID, getTodos } from './api/todos';
+import { Todo } from './types/Todo';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { ErrorNotification } from './components/ErrorNotification';
+
+export enum FilterType {
+  All = 'all',
+  Active = 'active',
+  Complited = 'complited',
+}
+
+function getVisibleTodos(todos: Todo[], filter: string) {
+  let newTodosList = [...todos];
+
+  if (filter) {
+    switch (filter) {
+      case FilterType.Active:
+        newTodosList = newTodosList.filter(todo => !todo.completed);
+        break;
+      case FilterType.Complited:
+        newTodosList = newTodosList.filter(todo => todo.completed);
+        break;
+      case FilterType.All:
+      default:
+        break;
+    }
+  }
+
+  return newTodosList;
```

Comment:
please try to get rid of let
can we return immediately and pass switch case in callback?